### ISSUE TITLE
Assignment 1

### ISSUE THREAD

**Issue by JoeBlanc** [JoeBlanc/df-frontend-2023#1]
link demo: https://df-frontend-2023-gamma-nine.vercel.app/
link repo: https://github.com/JoeBlanc/df-frontend-2023.git

**Comment by ngolapnguyen** [comment]
Good work 👍 

## Requirements
- [x] Render correct table elements
- [x] Create book
- [x] Delete book
- [x] Search book by title
- [x] Semantic HTML
- [x] Standard use of CSS and JavaScript
- [ ] Store data in LocalStorage

Final result: ✅ passed 70% of requirements

## Feedbacks

1. Minor stuff but there are a lot of typo... "boonkstore", "consfirmed", etc. Should pay more attention to this

2. Adding new book when filtering (searching) will render the whole book list. Should have a logic to either:
    1. Reapply the filter, OR
    2. Clear out the filter after adding new book, OR
    3. Add a new logic in the `buildTable` method to always take the filter into account when rendering

3. Naming could be better. A name should be concise, unambiguous and clear to the reader what it represents. For example, there are some naming improvements we can make in your code:
   1. `handleAddEvent` -> `handleAddEventListeners`, or simply `addEventListeners` (even though it's still a bit generic, but considering the scope of the assignment it's fine)
   2. `addBook` and `handleDeleteBook` should align on the naming pattern. Either it's `addBook` and `deleteBook`, or `handleAddBook` and `handleDeleteBook`. Be consistent
   3. `filter` -> `filterBooks`
   4. `open-modal-btn` -> `open-add-book-modal-btn`
   5. ...

In short, try to be precise & think how other programmers can look at your code & understand the flow of it. Naming is a small thing, but it's actually [one of the hardest problems in programming](https://makimo.com/blog/scientific-perspective-on-naming-in-programming/) & can directly affect your code quality & team-work.

**Comment by JoeBlanc** [comment]
I'm really impressed by all the details you mentioned.
Will try to improve and follow your suggestions. 
Thank you very much for your review.!
